### ISSUE TITLE
[US - DEV] As MILES HARDIN, I want to delete my idea page because I decided I don’t want to move forward with it

### ISSUE THREAD

**Issue by JulieMass** [dev-launchers/dev-launchers-platform#838]

**Comment by dbradham** [comment]
@JulieMass What happens if Miles clicks on the X or outside of the modal after deleting the idea? Does he just stay on the workshopping page of a deleted idea? Or do we have a page we want to redirect hm to?

**Comment by JulieMass** [comment]
> @JulieMass What happens if Miles clicks on the X or outside of the modal after deleting the idea? Does he just stay on the workshopping page of a deleted idea? Or do we have a page we want to redirect hm to?

This is an excellent question Dave. Thanks for pointing this out! I would say he should be redirected to the ideas page. I'll include this in the acceptance criteria. Please @Nouri-Anouar let me know if you agree with this as well.

**Comment by JulieMass** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/IdeaSpace-Team-6350185d9843157427cc900b/issues/dev-launchers/dev-launchers-platform/838?planning-poker) with Zenhub @Nouri-Anouar @JulieMass @dbradham @jordanmcousins @alicefate @RETprojects

**Comment by dbradham** [comment]
@alicefate do you know if we can delete these with the current backend?

**Comment by alicefate** [comment]
> @alicefate do you know if we can delete these with the current backend?

yes. we can.

**Comment by JulieMass** [comment]
Please [add your planning poker estimate](https://app.zenhub.com/workspaces/IdeaSpace-Team-6350185d9843157427cc900b/issues/dev-launchers/dev-launchers-platform/838?planning-poker) with Zenhub @rashikashaw

**Comment by alicefate** [comment]
@dbradham @JulieMass Do we really delete the idea? Or just mark it as deleted which hidden from all users.

**Comment by JulieMass** [comment]
> @dbradham @JulieMass Do we really delete the idea? Or just mark it as deleted which hidden from all users.

(FYI @Nouri-Anouar )

hehe another good question. 

Yes, I think from the user perspective it should feel that it was REALLY deleted. It would be ethical for us to respect Miles wishes and if he wants his idea deleted we should do as much as we can to delete it. 

Now...I don't know how deep we can delete the idea for example from the back end data. This is something where Dave could help us more.

**Comment by alicefate** [comment]
@JulieMass @pyxld-kris @dbradham 
Double check about deleted idea:
ideaCard status is "deleted".
It won't be show on dashboard and browsepage.
If anyone try to go to workshopping page or edit page by url, it will show a modal window tell them this idea has been deleted, and send them back to previous page.

**Comment by JulieMass** [comment]
(+ @Nouri-Anouar so he is in the loop)

Yes, that's correct for me Iris! 👍 

Good catch there covering the scenario of someone trying to access by URL. It didn't occur to me! 

My only question is: instead of showing a modal window, shouldn't we be presenting an error 404 page (page not found)? And if so, maybe Dev Launcher's already has a 404 error page that we can reuse?

**Comment by alicefate** [comment]
@JulieMass That is a good idea, create reusable error 404 page. I create ticket #1005 for it.

**Comment by JulieMass** [comment]
Excellent! thank you Iris for creating the ticket. I'll add some details to it like description, acceptance criteria etc. and I'll add it to our next sprint priority list.  🙌

**Comment by dbradham** [comment]
Made this PR to address this feature breaking with Strapi migration-https://github.com/dev-launchers/dev-launchers-platform/pull/1771

This was the same issue as with archiving an idea

**Comment by JulieMass** [comment]
Can't close the ticket as I don't see the action buttons for "delete" and idea in the workshopping page in production.

<img width="1411" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBOGR5QkE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--b4c739cc2593784107e7243637a6d10d36fc4d2a/image.png" alt="image.png" />


(CC. @dbradham)

**Comment by JulieMass** [comment]
Hey @dbradham, 

I'm sending this ticket back to the QA column as from our chat on Discord, it seems like this feature isn't fully done yet (the deleted ideas are still showing up in pages like browse) , so we can't close the ticket just yet. 

Let me know please if we need to pass this off to a different developer to get the work finished, so we can finally close this thing out. Just want to make sure we get it all sorted!

Thank you!

**Comment by dbradham** [comment]
@JulieMass yes, let's assign to another developer. Could you please add the expected behaviors for the idea browse and dashboard pages to the acceptance criteria of this ticket?

**Comment by JulieMass** [comment]
> @JulieMass yes, let's assign to another developer. Could you please add the expected behaviors for the idea browse and dashboard pages to the acceptance criteria of this ticket?

@dbradham Yes, acceptance criteria updated! Can you please take a look and make sure the acceptance criteria covers everything? I want to make sure I didn't miss anything. 

Also, if the AC looks good, could you please move it back to the product backlog column? In the meantime, I'm going to make a note that this needs to be reassigned to another developer in future sprints. Not sure why Iris’s work got lost here as I remember she worked on all this functionalities, maybe the strapi migration?🤷 

Anyway, thanks a bunch!
